### PR TITLE
feat: add docs params to proxy and mesh resources

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

Proxy resources like `Dataplane`, `ZoneIngress` and `ZoneEgress`,
as well as `Mesh` api resources are now marked with doc params,
which will allow us to generate the reference docs automatically.

Blocked by: https://github.com/kumahq/protoc-gen-kumadoc/pull/8 and pushing `0.2.0` tag for `protoc-gen-kumadoc` after the PR will be merged

### Full changelog

n/a

### Issues resolved

n/a

### Documentation

n/a

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: api/mesh/v1alpha1/dataplane.proto [hunk 1]
```diff
@@ -125,24 +135,25 @@ message Dataplane {
...
+      GatewayType type = 2 [ (doc.required) = true ];
     }
 
     // TransparentProxying describes configuration for transparent proxying.
     message TransparentProxying {
 
       // Port on which all inbound traffic is being transparently redirected.
       uint32 redirect_port_inbound = 1
-          [ (validate.rules).uint32 = {lte : 65535} ];
+          [ (validate.rules).uint32 = {lte : 65535}, (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/dataplane.proto:152]: This has a good default as well no?

File: api/mesh/v1alpha1/dataplane.proto [hunk 2]
```diff
@@ -125,24 +135,25 @@ message Dataplane {
...
 
       // Port on which all outbound traffic is being transparently redirected.
       uint32 redirect_port_outbound = 2
-          [ (validate.rules).uint32 = {lte : 65535} ];
+          [ (validate.rules).uint32 = {lte : 65535}, (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/dataplane.proto:156]: same

File: api/mesh/v1alpha1/dataplane.proto [hunk 3]
```diff
@@ -163,7 +174,7 @@ message Dataplane {
     Gateway gateway = 3;
 
     // Inbound describes a list of inbound interfaces of the dataplane.
-    repeated Inbound inbound = 1;
+    repeated Inbound inbound = 1 [ (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/dataplane.proto:177]: I think in the case of a gateway there's no inbound

File: api/mesh/v1alpha1/mesh.proto [hunk 4]
```diff
@@ -95,18 +102,18 @@ message Mesh {
...
-  string type = 2;
+  string type = 2 [ (doc.required) = true ];
 
   // DpCert defines settings for certificates generated for Dataplanes
   message DpCert {
     // Rotation defines rotation settings for Dataplane certificate
     message Rotation {
       // Time after which generated certificate for Dataplane will expire
-      string expiration = 1;
+      string expiration = 1 [ (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/mesh.proto:116]: Safe default

File: api/mesh/v1alpha1/mesh.proto [hunk 5]
```diff
@@ -119,7 +126,7 @@ message CertificateAuthorityBackend {
   DpCert dpCert = 3;
 
   // Configuration of the backend
-  google.protobuf.Struct conf = 4;
+  google.protobuf.Struct conf = 4 [ (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/mesh.proto:129]: Conf is not always required

File: api/mesh/v1alpha1/mesh.proto [hunk 6]
```diff
@@ -215,40 +222,40 @@ message ZipkinTracingBackendConfig {
...
   // Name of the backend, can be then used in Mesh.logging.defaultBackend or in
   // TrafficLogging
-  string name = 1;
+  string name = 1 [ (doc.required) = true ];
 
-  // Format of access logs. Placehodlers available on
+  // Format of access logs. Placeholders available on
   // https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log
-  string format = 2;
+  string format = 2 [ (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/mesh.proto:240]: Safe default aswell

File: api/mesh/v1alpha1/zone_ingress.proto [hunk 7]
```diff
@@ -28,26 +35,26 @@ message ZoneIngress {
...
+    string advertisedAddress = 2 [ (doc.required) = true ];
 
     // Port of the inbound interface that will forward requests to the service.
-    uint32 port = 3;
+    uint32 port = 3 [ (doc.required) = true ];
 
     // AdvertisedPort defines port on which ZoneIngress is accessible to other
     // Kuma clusters.
-    uint32 advertisedPort = 4;
+    uint32 advertisedPort = 4 [ (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/zone_ingress.proto:49]: same optional

File: api/mesh/v1alpha1/zone_ingress.proto [hunk 8]
```diff
@@ -28,26 +35,26 @@ message ZoneIngress {
 
   message Networking {
     // Address on which inbound listener will be exposed
-    string address = 1;
+    string address = 1 [ (doc.required) = true ];
 
     // AdvertisedAddress defines IP or DNS name on which ZoneIngress is
     // accessible to other Kuma clusters.
-    string advertisedAddress = 2;
+    string advertisedAddress = 2 [ (doc.required) = true ];
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply api/mesh/v1alpha1/zone_ingress.proto:42]: it's not: https://github.com/kumahq/kuma/blob/e2f56dda41a08bd3743bcccbf99d937fdd0403f8/pkg/core/resources/apis/mesh/zone_ingress_validator.go#L22-L23

File: api/mesh/v1alpha1/zone_ingress.proto (hunk 7 above)
Comment (bartsmykla) [reply api/mesh/v1alpha1/zone_ingress.proto:49]: it's not: https://github.com/kumahq/kuma/blob/e2f56dda41a08bd3743bcccbf99d937fdd0403f8/pkg/core/resources/apis/mesh/zone_ingress_validator.go#L25-L27

File: api/mesh/v1alpha1/mesh.proto (hunk 5 above)
Comment (bartsmykla) [reply api/mesh/v1alpha1/mesh.proto]: addressed

File: api/mesh/v1alpha1/dataplane.proto (hunk 3 above)
Comment (bartsmykla) [reply api/mesh/v1alpha1/dataplane.proto]: addressed

File: api/mesh/v1alpha1/mesh.proto [hunk 9]
```diff
@@ -52,7 +59,7 @@ message Mesh {
   Metrics metrics = 4;
 
   // Networking settings of the mesh
-  Networking networking = 5;
+  Networking networking = 5 [ (doc.required) = true ];
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply api/mesh/v1alpha1/mesh.proto]: addressed

File: api/mesh/v1alpha1/mesh.proto (hunk 6 above)
Comment (bartsmykla) [reply api/mesh/v1alpha1/mesh.proto]: addressed

File: api/mesh/v1alpha1/dataplane.proto [hunk 10]
```diff
@@ -71,18 +79,20 @@ message Dataplane {
       // ServiceProbe defines parameters for probing service's port
       message ServiceProbe {
         // Interval between consecutive health checks.
-        google.protobuf.Duration interval = 1;
+        google.protobuf.Duration interval = 1 [ (doc.required) = true ];
 
         // Maximum time to wait for a health check response.
-        google.protobuf.Duration timeout = 2;
+        google.protobuf.Duration timeout = 2 [ (doc.required) = true ];
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply api/mesh/v1alpha1/dataplane.proto]: addressed

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**